File: flutter_boilerplate_dash/lib/module_container/request/container_filter_request.dart
class ContainerFilterRequest {
  String? type;
  String? containerNumber;
  String? status;

  int? providedBy;
  int? shipperID;
  int? consigneeID;
  int? specificationID;
  int? shipmentID;
  int? clientUserID;
  int? containerID;

  bool? isExternalWarehouse;
  bool? withoutNumber;
  bool? isRequested;

  ContainerFilterRequest({
    this.type,
    this.containerNumber,
    this.status,
    this.providedBy,
    this.consigneeID,
    this.shipperID,
    this.specificationID,
    this.isExternalWarehouse,
    this.shipmentID,
    this.containerID,
    this.clientUserID,this.isRequested,this.withoutNumber

  });

  Map<String, dynamic> toJson() {
    return {
      'type': type ?? '',
      'status': status ?? '',
      'shipperID': shipperID ?? 0,
      'consigneeID': consigneeID ?? 0,
      'providedBy': providedBy ?? 0,
      'specificationID': specificationID ?? 0,
      'containerNumber':containerNumber ??'',
      'isExternalWarehouse':isExternalWarehouse,
      'shipmentID':shipmentID,
      'clientUserID':clientUserID,
      'id':containerID,
      'isRequested':isRequested,
      'withoutNumber':withoutNumber
    };
  }
}

File: backend/src/Request/ContainerFilterRequest.php
<?php

namespace App\Request;

class ContainerFilterRequest
{
    private $specificationID;

    private $containerNumber;

    private $status;

    private $type;

    private $providedBy;

    private $shipperID;

    private $consigneeID;

    private $isExternalWarehouse;

    private $shipmentID;

    private $clientUserID;

    private $withoutNumber;

    private $isRequested;

    public function getSpecificationID()
    {
        return $this->specificationID;
    }

    public function getType()
    {
        return $this->type;
    }

    public function getContainerNumber()
    {
        return $this->containerNumber;
    }

    public function getConsigneeID()
    {
        return $this->consigneeID;
    }

    public function getProvidedBy()
    {
        return $this->providedBy;
    }

    public function getShipperID()
    {
        return $this->shipperID;
    }

    public function getStatus()
    {
        return $this->status;
    }

    public function getIsExternalWarehouse()
    {
        return $this->isExternalWarehouse;
    }

    public function getShipmentID()
    {
        return $this->shipmentID;
    }

    public function getClientUserID()
    {
        return $this->clientUserID;
    }

    public function getWithoutNumber()
    {
        return $this->withoutNumber;
    }

    public function getIsRequested()
    {
        return $this->isRequested;
    }

}